Sources/MessagingPush/MessagingPush.swift

```diff
@@ -0,0 +1,122 @@
+import CioTracking
+import Foundation
+
+/**
+ Swift code goes into this module that are common to *all* of the Messaging Push modules (APN, FCM, etc).
+ So, performing an HTTP request to the API with a device token goes here.
+  */
+open class MessagingPush {
+    
+    public private(set) static var instance = MessagingPush(customerIO: CustomerIO.instance)
+
+    private let customerIO: CustomerIO!
+    private let httpClient: HttpClient
+
+    private var credentials: SdkCredentials? {
+        customerIO.credentials
+    }
+
+    private var sdkConfig: SdkConfig {
+        customerIO.sdkConfig
+    }
+    
+    public var deviceToken: Data?
```

Comment:
```suggestion
    @Atomic public var deviceToken: Data?
```

make it thread safe since we may be accessing it in many places.